framework/sandstone.cpp

```diff
@@ -1606,7 +1609,7 @@ static void wait_for_children(ChildrenList &children, int *tc, const struct test
             children.results[i].usage = usage;
             --children_left;
         }
-        return 0;
+        return caughtSignal;
```

Comment:
This is behavioral change. The execution **will not** continue after ALARM/SEG FAULT/ABORT/etc.
Instead of this additional variable, we can check the overall status and react on Killed/Interrupted with appropriate exit code (if we're allowed to change old Ctrl-C behavior).

Reply:
No, that's not the case. Testing does show it works properly. Just run the `selftest_sigsegv` and you'll see it gets retried properly.

This `caughtSignal` here only applies if the parent process gets a signal and only the SIGHUP, SIGINT, SIGTERM, and SIGPIPE signals (`terminationSignals` in unix/signals.cpp). It does not apply to crashing signals or SIGALRM received by the parent process and it does not apply to any signals received by the child processes either.

Reply:
Sorry, I was wrong, you're right.
Handling here is limited to handle signals from "external" world (sent by the operator or OS, e.g. service termination, Ctrl-C, Ctrl-|, etc), not the ones "automatically sent" (on "critical" issues in test execution).